### PR TITLE
Push cloud-config.yaml into autoscaler sidecar container

### PR DESCRIPTION
**Pull request by addyess**
Cannot be pushed until a new oci-image is created

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -28,4 +28,4 @@ resources:
   juju-autoscaler-image:
     type: oci-image
     description: OCI Image for kubernetes-autoscaler with a juju cloud provider.
-    upstream-source: rocks.canonical.com/cdk/cluster-autoscaler-juju:dev-2.9.22-build4
+    upstream-source: rocks.canonical.com/cdk/cluster-autoscaler-juju:dev-2.9.22-build??
```
Comment (addyess) [inline metadata.yaml]: DO NOT MERGE without a proper `upstream-source`

File: src/autoscaler.py [hunk 2]
```diff
@@ -75,6 +87,7 @@ def authorize(self, container):
         container.push(*self.accounts_file, make_dirs=True, **self.root_owned)
         container.push(*self.controller_file, make_dirs=True, **self.root_owned)
         container.push(*self.models_file, make_dirs=True, **self.root_owned)
```
Comment (addyess) [inline src/autoscaler.py]: i'd expect you'd want to remove these @stonepreston?

Comment (stonepreston) [reply src/autoscaler.py]: yes if we get HA working with the [simple connector PR](https://github.com/charmed-kubernetes/autoscaler/pull/6) then we would no longer need to build up these client side files and push them into the container

File: metadata.yaml (hunk 1 above)
Comment (stonepreston) [reply metadata.yaml]: New image is now available [here](https://rocks.canonical.com/v2/cdk/cluster-autoscaler-juju/tags/list). The new tag is dev-2.9.latest-build0 as the juju api is using the tip of the main branch that is not part of a juju release yet.

File: tests/data/pebble_cfg_minimum/cloud-config.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+Cacert: |-
```
Comment (stonepreston) [inline tests/data/pebble_cfg_minimum/cloud-config.yaml]: I changed the yaml identifiers to be more yamly as requested. The yaml identifiers should now be:
```
user
password
endpoints
ca-cert
```

File: src/autoscaler.py [hunk 4]
```diff
@@ -3,20 +3,32 @@
...
+
+
+ControllerData = TypedDict(
+    "ControllerData",
+    {"api-endpoints": list[str], "uuid": str, "ca-cert": str},
+)
+Secrets = TypedDict("Secrets", {"user": str, "password": str})
+ModelData = TypedDict("ModelData", {"models": dict, "current-model": str})
+CloudConfig = TypedDict(
+    "CloudConfig", {"Username": str, "Password": str, "Endpoint": list[str], "Cacert": str}
```
Comment (stonepreston) [inline src/autoscaler.py]: I changed the yaml identifiers for the cloud  config to be more yamly as requested. The yaml identifiers should now be:
```
user
password
endpoints
ca-cert
```

File: tests/unit/test_charm.py [hunk 5]
```diff
@@ -127,27 +127,11 @@ def test_juju_autoscaler_pebble_ready_after_config_minimal(
     text = Path(testdata, "layer.yaml").read_text()
     assert plan.to_dict() == yaml.safe_load(text)
 
-    cli_path = "/root/.local/share/juju"
+    cli_path = "/config"
```
Comment (stonepreston) [inline tests/unit/test_charm.py]:
```suggestion
    config_path = "/config"
```

File: tests/unit/test_charm.py [hunk 6]
```diff
@@ -127,27 +127,11 @@ def test_juju_autoscaler_pebble_ready_after_config_minimal(
...
     files = container.list_files(cli_path)
```
Comment (stonepreston) [inline tests/unit/test_charm.py]:
```suggestion
    files = container.list_files(config_path)
```

**Review by stonepreston — CHANGES_REQUESTED** [review]
small nit with a variable name referencing cli

**Review by stonepreston — APPROVED** [review]
LGTM

**Pull request merged**